flake.nix

```
@@ -73,6 +75,7 @@
               libsodium-bindings = commonOverrides (hself.callCabal2nix "libsodium-bindings" (gitignoreSource ./libsodium-bindings) { });
               sel = commonOverrides (hself.callCabal2nix "sel" (gitignoreSource ./sel) { });
               text-display = markUnbroken hsuper.text-display;
+              base16 = inputs.base16;
```

base16 should be a derivation that contains a haskell package, more specifically it should not be a store path to the source of base qy as it is currently. 
```suggestion
              base16 = hself.callCabal2nix "base16" (gitignoreSource inputs.base16);
```

---

I see